fix: remove waitForElementNotPresent and improve assertElementNotPresent

HowardBraham opened the pull request:
## **Description**

**The problem:** `waitForElementNotPresent` would pass immediately (which was sometimes TOO fast), and `assertElementNotPresent` always waited at least 10 seconds before passing

**The solution:**
- Added 3 named parameters to `assertElementNotPresent`
  - findElementGuard - A rawLocator to perform a findElement and act as a guard
  - waitAtLeastGuard - The minimum milliseconds to wait before passing
  - timeout - The maximum milliseconds to wait before failing
- Removed `waitForElementNotPresent` from helpers.js
- Converted all instances of  `waitForElementNotPresent` to `assertElementNotPresent`
- For each statement that was originally an `assertElementNotPresent`, I put in a proper guard, and commented on that guard.  Each time I did this, it sped up the test by 9-10 seconds.
- While I was in there, I realized that `pin account by clicking the pin button` and `unpin account by clicking the unpin button` could be combined into a single test

I think this is a start, but I doubt it's complete... I bet there are more `assertElementNotPresent` statements out there that need guards, and cause flaky tests because they don't have guards.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23108?quickstart=1)

## **Related issues**
## **Manual testing steps**
## **Screenshots/Recordings**
## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

HowardBraham committed:
fix: remove assertElementNotPresent and improve waitForElementNotPresent

DDDDDanica commented on test/e2e/webdriver/driver.js:
```diff
@@ -259,9 +258,47 @@ class Driver {
+   * @param {number} waitAtLeast - The minimum milliseconds to wait before passing
+   * @param {number} timeout - The maximum milliseconds to wait before failing
+   */
+  async waitForElementNotPresent(
```
Nit: After checking your implementation, i think we can even pass another parameter as the guard selector, which is optional. So we can execute `findElement` of guard and then execute the not present await. what's your thoughts?

HowardBraham replied:
Yeah I considered that, do you think it would be better?  Would it use a mix of positional and named parameters like this?

```
async waitForElementNotPresent(
  rawLocator,
  { findElementGuard = null, waitAtLeastGuard = 0, timeout = this.timeout },
) {
```

DDDDDanica replied:
Looks clear to me to separate them like that !

HowardBraham commented:
Harika suggested that the name for the function should be `assertElementNotPresent`, not `waitForElementNotPresent`.  I think that makes sense, I'll change it.

HowardBraham committed:
changed all the new calls to assertElementNotPresent

HowardBraham committed:
changed all previously untouched to assertElementNotPresent

HowardBraham committed:
update-network.spec.ts improvements

HowardBraham committed:
fixing send-eth.spec.js

HowardBraham committed:
lint fixes

HowardBraham commented:
@DDDDDanica and @hjetpoluru this is ready for you guys to review again.  I added the `findElementGuard` parameter and renamed all to `assertElementNotPresent`.

chloeYue reviewed (dismissed):
LGTM! Passing the minimum wait time as a parameter, and use of `findElementGuard` will significantly improve how we wait for an element to be absent.

HowardBraham committed:
undid one of the update-network.spec.ts changes

HowardBraham committed:
moved a few `if (process.env.MULTICHAIN)` statements upwards

hjetpoluru committed:
Merge branch 'develop' into e2e/wait-for-not-present

hjetpoluru reviewed (approved):
@HowardBraham Great job on enhancing the library function assertElementNotPresent and conducting additional cleanup, including comments, particularly with the multichain flag at the beginning of the test. Your attention to detail is spot on and truly appreciated!

bowensanders reviewed (approved):
LGTM ✅
